tests/integration/mirrorpeer_status_test.go

```diff
@@ -0,0 +1,191 @@
+package integration_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	fakeMirrorPeerName = "fake-mirror-peer"
+	testCluster1       = "test-provider-cluster1"
+	testCluster2       = "test-provider-cluster2"
+	testSC1            = "test-storagecluster-1"
+	testSC2            = "test-storagecluster-2"
+	testSCNamespace    = "test-namespace"
+)
+
+func GetFakeMirrorPeerReconciler() (*controllers.MirrorPeerReconciler, *multiclusterv1alpha1.MirrorPeer, error) {
+
+	fakeMirrorPeer := GetFakeMirrorPeer()
+
+	scheme := runtime.NewScheme()
+	err := multiclusterv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		return nil, nil, err
+	}
+	err = clusterv1.AddToScheme(scheme)
+	if err != nil {
+		return nil, nil, err
+	}
+	err = multiclusterv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		return nil, nil, err
+	}
+	err = addonapiv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		return nil, nil, err
+	}
+	err = corev1.AddToScheme(scheme)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	fakeMirrorPeerReconciler := &controllers.MirrorPeerReconciler{
+		Client: fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(fakeMirrorPeer).Build(),
+		Scheme: scheme,
+	}
+
+	return fakeMirrorPeerReconciler, fakeMirrorPeer, nil
+}
+
+func GetFakeMirrorPeer() *multiclusterv1alpha1.MirrorPeer {
+	return &multiclusterv1alpha1.MirrorPeer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fakeMirrorPeerName,
+		},
+		Spec: multiclusterv1alpha1.MirrorPeerSpec{
+			Items: []multiclusterv1alpha1.PeerRef{
+				{
+					ClusterName: testCluster1,
+					StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+						Name:      testSC1,
+						Namespace: testSCNamespace,
+					},
+				},
+				{
+					ClusterName: testCluster2,
+					StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+						Name:      testSC2,
+						Namespace: testSCNamespace,
+					},
+				},
+			},
+		},
+	}
+}
```

Comment:
@aruniiird also created a similar function. This is exactly why I suggest defining this fake/mock/default value locally for each test.

Reply:
created a mock object.